abr_control/controllers/osc.py

```diff
@@ -133,16 +136,20 @@ def generate(self, q, dq,
             u_task[:3] = -self.kv * (dx - target_vel -
                                      np.clip(sat / scale, 0, 1) *
                                      -self.lamb * scale * x_tilde)
+            # low level signal set to zero
+            u = 0.0
```

Comment:
why do we have `u+=` when we don't use `vmax`, but when we do we reset to zero every step `u=0` ?

Reply:
i think in both cases we're using `u =`, no? which line are you seeing `u+= `? 

my idea for moving the `u` initialization in here was that we can account for velocity compensation in joint space or not right here rather than having to have a separate if statement to check if we used `vmax` later on or not

Reply:
we have `u+=` on line 164. It will lead to different behaviour with vmax since we won't actually be adding to the previous value since it gets reset to zero.

Reply:
so i see `u` getting set in line 140 with `vmax` and 145 otherwise, and then adding to `u` in 164. is there another spot or are these the three we're talking about?